src/api/routes/myGroups.ts

```diff
@@ -12,43 +12,66 @@ import { createMeeting } from "../TencentMeeting";
 import Transcript from "../database/models/Transcript";
 import moment from 'moment';
 import { noPermissionError, notFoundError, zGroupCountingTranscripts } from "./groups";
+import OngoingMeetingCount from "api/database/models/OngoingMeetingCount";
+import apiEnv from "api/apiEnv";
 
 export function meetingLinkIsExpired(meetingLinkCreatedAt : Date) {
-  // meeting is valid for 31 days after start time
+  // meeting is valid for 30 days after start time
   // check if the link is created within 30 days
   return moment() > moment(meetingLinkCreatedAt).add(30, 'days');
 }
 
 const myGroups = router({
 
-  generateMeetingLink: procedure
+  joinMeetingLink: procedure
   .use(authUser())
   .input(z.object({ groupId: z.string() }))
   .mutation(async ({ ctx, input }) => {
+    let meetingLink : string | null;
+    let meetingId: string | null;
     const group = await Group.findByPk(input.groupId, {
       include: [GroupUser]
     });
     if (!group) throw notFoundError(input.groupId);
     // Only meeting members have access to this method.
     if (!group.groupUsers.some(gu => gu.userId === ctx.user.id)) throw noPermissionError(input.groupId);
 
-    if (group.meetingLink && !meetingLinkIsExpired(group.updatedAt)) {
-      return group.meetingLink;
-    }
+    if (group.meetingLinkCreatedAt && !meetingLinkIsExpired(group.meetingLinkCreatedAt)) {
+      meetingLink = group.meetingLink;
+      meetingId = group.meetingId;
+    } else {
 
-    const now = Math.floor(Date.now() / 1000);
-    const res = await createMeeting(group.id, now, now + 3600);
-    invariant(res.meeting_info_list.length === 1);
+      const now = Math.floor(Date.now() / 1000);
+      const res = await createMeeting(group.id, now, now + 3600);
+      invariant(res.meeting_info_list.length === 1);
 
-    const meetingLink = res.meeting_info_list[0].join_url;
+      meetingLink = res.meeting_info_list[0].join_url;
+      meetingId = res.meeting_info_list[0].meeting_id;
 
-    await Group.update({
-      meetingLink,
-    }, {
-      where: {
-        id: group.id,
-      }
-    });
+      await Group.update({
+        meetingLink,
+        meetingId,
+        meetingLinkCreatedAt: moment.unix(now).toDate(),
+      }, {
+        where: {
+          id: group.id,
+        }
+      });
+    }
+
+    const ongoingMeeting = await OngoingMeetingCount.findByPk(apiEnv.TM_ADMIN_USER_ID);
+    // Check if the meeting id of the ongoingMeeting matches the meeting id of the group tries to join
+    // The group user will allow to join the meeting if matched
+    // otherwise return null to warn that user cannot join right now due to other ongoing meetings
+    if (ongoingMeeting && ongoingMeeting.count !== 0) {
+      return ongoingMeeting.meetingId === meetingId ? meetingLink : null;
+    }
+
+    OngoingMeetingCount.upsert({
+      TMAdminUserId: apiEnv.TM_ADMIN_USER_ID,
+      count: 1,
+      meetingId,
+    })
 
```

Comment:
extract this entire block into a function `enforceMeetingQuota(meetingId: string): void`, which throws a TRPCError with a specific type when quota is reached. Don't return null to indicate errors.